### ISSUE TITLE
numerical styles with second-field-align="flush"

### ISSUE THREAD

**Issue by njbart** [jgm/pandoc-citeproc#29]
Using numerical styles such as nature.csl which contain the csl directive `second-field-align="flush"` in `<bibliography ...>`, the bibliography is expected to be rendered like a numbered list. 

(There are also a few styles with `second-field-align="margin"`, see list at https://gist.github.com/gjuggler/8222414, but since I don't think pandoc can handle stuff in margins these are probably best formatted just like the ones with `second-field-align="flush"`.)

In addition, there should be no space before the superscripted citation numbers. (Citation numbers always appearing _before_ final punctuation seems to be intentional, however, as a look at a number of recent "Nature" articles shows, and the fact that pandoc ignores page numbers is fine since these just aren't supposed to be used with such styles.)

The "et al" issue has been reported before, see https://github.com/jgm/pandoc-citeproc/issues/27.

There are a few other unexpected details, such as missing location and "ed.", but these are csl style errors.

```
pandoc -s --filter pandoc-citeproc -t markdown-citations <<EOT

Lorem [@arj1]. Ipsum [@boo1, 12-34; @inc1].

# References {-}

---
csl: nature.csl
references:
- page: 271-279
  title: By rail and stage to Galena
  id: inc1
  issued:
    date-parts:
    - - 1968
  author:
  - given: Elizabeth F. L.
    family: Ellet
  container-title: "Prairie state: Impressions of Illinois, 1673–1967, by travelers
    and other observers"
  editor:
  - given: Paul M.
    family: Angle
  publisher: University of Chicago Press
  type: chapter
  publisher-place: Chicago
- title-short: Freakonomics
  title: 'Freakonomics: A rogue economist explores the hidden side of everything'
  id: boo1
  issued:
    date-parts:
    - - 2005
  author:
  - given: Steven D.
    family: Levitt
  - given: Stephen J.
    family: Dubner
  publisher: William Morrow
  type: book
  publisher-place: New York
- title-short: Codes, costs, and critiques
  issue: 1
  page: 19-35
  title: "Codes, costs, and critiques: The organization of information in *Library
    Quarterly*, 1931–2004"
  volume: 76
  id: arj1
  issued:
    date-parts:
    - - 2006
  author:
  - given: Hope A.
    family: Olson
  container-title: Library Quarterly
  type: article-journal
  DOI: 10.1086/504343
...
EOT
```

Actual output:

```
Lorem ^1^. Ipsum ^2,3^.

<div class="references">

References {#references .unnumbered}
==========

1.Olson, H. A. *et al.* Codes, costs, and critiques: The organization of
information in *Library Quarterly*, 1931–2004. *Library Quarterly*
**76,** 19–35 (2006).

2.Levitt, S. D. & Dubner, S. J. *et al.* *Freakonomics: A rogue
economist explores the hidden side of everything*. (William Morrow,
2005).

3.Ellet, E. F. L. *et al.* in *Prairie state: Impressions of Illinois,
1673–1967, by travelers and other observers* (Angle, P. M.) 271–279
(University of Chicago Press, 1968).

</div>
```

Expected output:

```
Lorem^1^. Ipsum^2,3^.

<div class="references">

References {#references .unnumbered}
==========

1.  Olson, H. A. *et al.* Codes, costs, and critiques: The organization of
    information in *Library Quarterly*, 1931–2004. *Library Quarterly*
    **76,** 19–35 (2006).

2.  Levitt, S. D. & Dubner, S. J. *et al.* *Freakonomics: A rogue
    economist explores the hidden side of everything*. (William Morrow,
    2005).

3.  Ellet, E. F. L. *et al.* in *Prairie state: Impressions of Illinois,
    1673–1967, by travelers and other observers* (Angle, P. M.) 271–279
    (University of Chicago Press, 1968).

</div>
```

**Comment by jgm** [comment]
Note to self: we can get at this where `s` is a style via

    lookup "second-field-align" $ maybe [] (bibOptions) $ biblio s

If the value is `Just "margin"`, we should insert a line break after the citation number.
If it is `Just "flush"`, we should (for now) insert a space after the citation number.

**Comment by jgm** [comment]
I have implemented a fix for this that isn't perfect, but is better than what we had.
We now at least get a space after the citation number if second-field-align is flush.
If it's margin, we get a line break.
I don't see a good way of doing better without more major changes.

**Comment by jgm** [comment]
> If it's margin, we get a line break.

Sorry, this was my misunderstanding.  I've changed this so a space, rather than a line break, is inserted.

**Comment by januz** [comment]
I just started playing around with creating a custom CSL style and also ran into this problem (`second-field-align: flush` not being applied); also `entry-spacing` does not seem to be applied.

Are those still known issues/bugs and/or is there something I can do to achieve the desired layout in HTML output? Thank you!

**Comment by njbart** [comment]
Fixed in pandoc 2.11. Closing.
